src/bms/parse/header.rs

```diff
@@ -34,6 +34,66 @@ pub struct Bmp {
     pub transparent_color: Argb,
 }
 
+/// A definition for #@BGA command.
+#[derive(Debug, Clone, PartialEq)]
+pub struct AtBgaDef {
+    /// The object ID.
+    pub id: ObjId,
+    /// The source BMP object ID.
+    pub source_bmp: ObjId,
+    /// The top-left position for trimming.
+    pub trim_top_left: (i16, i16),
+    /// The size for trimming.
+    pub trim_size: (u16, u16),
+    /// The draw point position.
+    pub draw_point: (i16, i16),
+}
+
+/// A definition for #BGA command.
+#[derive(Debug, Clone, PartialEq)]
+pub struct BgaDef {
+    /// The object ID.
+    pub id: ObjId,
+    /// The source BMP object ID.
+    pub source_bmp: ObjId,
+    /// The top-left position for trimming.
+    pub trim_top_left: (i16, i16),
+    /// The bottom-right position for trimming.
+    pub trim_bottom_right: (i16, i16),
+    /// The draw point position.
+    pub draw_point: (i16, i16),
+}
+
+/// A definition for #EXRANK command.
+#[derive(Debug, Clone, PartialEq)]
+pub struct ExRankDef {
+    /// The object ID.
+    pub id: ObjId,
+    /// The judge level.
+    pub judge_level: JudgeLevel,
+}
+
+/// A definition for #EXWAV command.
+#[derive(Debug, Clone, PartialEq)]
+pub struct ExWavDef {
+    /// The object ID.
+    pub id: ObjId,
+    /// The pan of the sound. Also called volume balance.
+    /// Range: [-10000, 10000]. -10000 is leftmost, 10000 is rightmost.
+    /// Default: 0.
+    pub pan: i64,
+    /// The volume of the sound.
+    /// Range: [-10000, 0]. -10000 is 0%, 0 is 100%.
+    /// Default: 0.
+    pub volume: i64,
+    /// The frequency of the sound. Unit: Hz.
+    /// Range: [100, 100000].
+    /// Default: None.
+    pub frequency: Option<u64>,
```

Comment:
These options have own valid range and default value. How about adding new types for them such as `ExWavPan(i64)`, `ExWavDecay(i64)` and `ExWavPitch(u64)`?